ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWT.scala

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.ledger.api.auth
+
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.jwt.JwtVerifier
+import com.digitalasset.ledger.api.auth.AuthServiceJWT.Error
+import io.grpc.Metadata
+import org.slf4j.{Logger, LoggerFactory}
+import spray.json._
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/** The JWT token payload used in [[AuthServiceJWT]]
+  *
+  * For forward/backward compatibility reasons, all fields are optional.
+  *
+  * The information in this token may be used to fill in missing data in gRPC ledger API requests.
+  */
+case class AuthServiceJWTPayload(
+    ledgerId: Option[String],
+    participantId: Option[String],
+    applicationId: Option[String],
+    exp: Option[String],
+    admin: Option[Boolean],
+    actAs: Option[List[String]],
+    readAs: Option[List[String]]
+)
+
+object AuthServiceJWTProtocol extends DefaultJsonProtocol {
+  implicit val jwtPayloadFormat = jsonFormat7(AuthServiceJWTPayload.apply)
```

Comment:
In general I'm opposed to automatic derivation of codecs. Especially in this case, the JWT format needs to be defined in some sort of specification that is part of the documentation, and the implementation needs to be able to read that token. This decouples the format of the token from the representation we actually want to work with. It would allow us to avoid using `Option[List[String]]` in favor of just `List[String]` for the party related claims, or just `Boolean` instead of `Option[Boolean]` (the `None` case is really just `false`).